packages/svelte/src/compiler/phases/3-transform/client/visitors/shared/fragment.js

```diff
@@ -9,58 +10,85 @@ import { build_template_literal, build_update } from './utils.js';
  * (e.g. `{a} b {c}`) into a single update function. Along the way it creates
  * corresponding template node references these updates are applied to.
  * @param {SvelteNode[]} nodes
- * @param {(is_text: boolean) => Expression} expression
+ * @param {(is_text: boolean) => Expression} initial
  * @param {boolean} is_element
  * @param {ComponentContext} context
  */
-export function process_children(nodes, expression, is_element, { visit, state }) {
+export function process_children(nodes, initial, is_element, { visit, state }) {
 	const within_bound_contenteditable = state.metadata.bound_contenteditable;
+	let prev = initial;
+	let skipped = 0;
 
 	/** @typedef {Array<Text | ExpressionTag>} Sequence */
-
 	/** @type {Sequence} */
 	let sequence = [];
 
+	/** @param {boolean} check */
+	function get_node(check) {
+		if (skipped === 0) {
+			return prev(check);
+		}
+
+		return b.call(
+			'$.sibling',
+			prev(false),
+			(check || skipped !== 1) && b.literal(skipped),
+			check && b.true
+		);
+	}
+
+	/**
+	 * @param {boolean} check
+	 * @param {string} name
+	 */
+	function flush_node(check, name) {
+		const expression = get_node(check);
+		let id = expression;
+
+		if (id.type !== 'Identifier') {
+			id = b.id(state.scope.generate(name));
+			state.init.push(b.var(id, expression));
+		}
+
+		prev = () => id;
+		skipped = 1;
```

Comment:
why 1? Could use an explanatory comment

Reply:
yeah, this code makes me itch. i spent an unreasonable amount of time trying to make is `0` instead and remove the `skipped -= 1` below, but couldn't quite make things behave. it's a new day so will take another run at it

Reply:
ok i gave up — the cure would be worse than the disease. settled for adding comments instead